### DISCUSSION TITLE
How to enable `path` only ?

### DISCUSSION THREAD

**Discussion by 4l1fe** [bensadeh/tailspin#122] [Q&A] [answered]
I see no built-in way to colorize only `path` patrts of logs.

I would like to parse stdout for unix pathes only using `tspin`. Do i have to set up a config and write many settings for it? Is there a simplier way such as `cat stdout.txt | tspin --paths | less` ?

**Comment by bensadeh** [comment, marked as answer]
Hi, there is currently no flags for this functionality, and like you say, the way to do it today is to create a config where you disable all the highlight groups in the config toml as well as the keyword highlight groups.

Would you be able to share why you'd want to disable the rest of the highlight groups? I've put a lot of time into applying the highlights in such an order that will cause the least amount of interference with other highlight groups. Sometimes it is unavoidable, but other times a clash in highlighters might be considered a bug.

**Reply by 4l1fe** [reply]
> Would you be able to share why you'd want to disable the rest of the highlight groups? 

Hey, sure.

## Expectation
What do we need a colorizer for? To emphasize and highlight specific information within a large amount of data or a stream. For this reason, i would like to take a util that reduces the time spent on such the emphasizing.

When i found Tailspin, it attracted me by offering very practical features like _Highlights numbers, dates, IP-addresses, UUIDs, URLs and more_. Doest it save my time from writing regex and colorizing output? sure, it's great, i decided to take it.

But do i always need all the highligthing options that are all enabled by default? No.
What about opting in features to use for a specific case, such as seeing black color text lines and ,let's say, red,bold text for the paths?

## Given
At this point, Tailspin becomes a bit cluttered. For set it up to highlight only one option, i have to write multiple lines config when i could write a one-line command such as `tmux capture-pane -t1 -S- -p | tspin --paths red | less`

For now, i'm using `colout`, another really good colorizer. The Colout command looks like `tmux capture-pane -t1 -S- -p | colout -c '(\/.+?) ' red | less`

## Colout Drawbacks
Tailspin provides the features, i've already mentioned, numbers, paths, addresses,log levels...
The Colout interface is regexes. I prefer to think in terms of paths, numbers, but regexes. Tailspin wins here, it aligns more with human thinking.

## Tailspin Drawbacks
I could actually write many configs with various settings combinatios one time and make a few shell aliases.
```
alias tspinp=tspin --config-path /only-path-config.cfg
alias tspinph=tspin --config-path /only-path-http-config.cfg
alias tspinal=tspin --config-path /only-address-loglevel-config.cfg
...
```
But these aliases could be the commands with no configs if there are corresponding flags
```
stdout | tspin --paths
stdout | tspin --paths --http
stdout | tspin --addresses --loglevels
```

As you can see, having many configs and aliases makes Tailspin less handy compared to Colout. That's why i'm asking the quesiton.

**Reply by bensadeh** [reply]
Thanks for explaining your need in such detail. When it comes to adding flags to customize tailspin, it's something I am happy to add.

From my point of view, I'd like the options to be simple to use and understand. It would be great if the options were "symmetric" in the sense that you'd want just as easily disable a highlight group as enable it (and disable the rest, like in your question).

For example:
```console
# Only highlight paths and numbers, disable all other groups
tspin application.log --enable-paths --enable-numbers

# Disable highlighting paths and numbers, enable all other groups
tspin application.log --disable-paths --disable-numbers
```

Then you could leave the customizing of the groups to the actual `config.toml`.

**Reply by 4l1fe** [reply]
Sure, arguments naming you proposed definitely make sense. I didn't focus on it describing the problem about not having them.
